integration/tests/cook/util.py

```diff
@@ -333,14 +333,19 @@ def predicate(resp):
     return response.json()[0]
 
 
-def wait_for_sandbox_directory(cook_url, job_id, max_delay_ms=4000):
+def wait_for_sandbox_directory(cook_url, job_id):
     """
     Wait for the given job's sandbox_directory field to appear.
     Returns an up-to-date job description object on success,
     and raises an exception if the max_delay_ms wait time is exceeded.
     """
     job_id = unpack_uuid(job_id)
 
+    cook_settings = settings(cook_url)
+    cache_ttl_ms = get_in(cook_settings, 'agent-query-cache', 'ttl-ms')
+    sync_interval_ms = get_in(cook_settings, 'sandbox-syncer', 'sync-interval-ms')
+    max_delay_ms = min(4 * max(cache_ttl_ms, sync_interval_ms), 4 * 60 * 1000)
```

Comment:
One last question...is it possible to write a non-flaky integration test that would have failed before but pass now? I'm not sure it is, but it would be great to have.

Reply:
Let me see if I can make use of `mesos_mock`. An integration test might be hard as we use so few hosts that any of the other finishing jobs may trigger a sync.

Reply:
Added `test-sandbox-directory-population` in `scheduler.clj`.